src/SeleniumLibrary/keywords/cookie.py

```diff
@@ -55,24 +55,55 @@ def get_cookie_value(self, name):
 
     @keyword
     def get_cookie(self, name):
-        """Returns the cookie found with `name` in extended variable format.
-
-        If no cookie is found with `name`, this keyword fails.
+        """Returns a cookie object found with ``name``.
+
+        If no cookie is found with ``name``, keyword fails. The cookie object
+        contains details about the cookie. Attributes available in the object
+        are documented in the table below.
+        | = Attribute = |             = Explanation =                            |
+        | domain        | Specifies hosts to which the cookie will be sent       |
+        | expiry        | The maximum lifetime of the cookie as EPOCH            |
+        | httpOnly      | HttpOnly cookie cannot be accessed by client-side APIs |
+        | name          | The name of a cookie                                   |
+        | path          | Indicates a URL path, example /                        |
+        | secure        | Cookie will be send only by using secure connection    |
+        | value         | Value of the cookie                                    |
+        | full_info     | All the above attributes joined as string              |
+
+        New in SeleniumLibrary 3.0
+
+        Example:
+        | Add Cookie      | foo             | bar |
+        | ${cookie} =     | Get Cookie      | foo |
+        | Should Be Equal | ${cookie.value} | foo |
+        | Should Be Equal | ${cookie.name}  | bar |
         """
         cookie = self.browser.get_cookie(name)
         if cookie:
-            cookie_information = CookieInformation(cookie)
-            return cookie_information
+            return CookieInformation(
+                cookie.get('domain', None), cookie.get('expiry', None),
+                cookie.get('httpOnly', None), cookie['name'],
+                cookie.get('path', None), cookie.get('secure', None),
+                cookie['value'])
```

Comment:
No need to have e.g. `cookie.get('doman', None)` when `cookie.get('domain)` works exactly the same way. But do we even need that or could we just use `cookie['name']`? I don't see the spec saying anything that some of thee values would be optional: https://w3c.github.io/webdriver/webdriver-spec.html#cookies

If we can trust that the return value from `get_cookie` always have exactly the same content, we could actually use `CookieInformation(**cookie)`. That is a bit risky as it would break if any new information is added to the return value.

Reply:
That was the original implementation but atleast in Windows all the keys are not always returned. Then it could lead to situations where the advanced variables are not always populated when running the same test by using different browser and/or different OS. 

I agree with the `cookie.get` comment.

Reply:
Done